### ISSUE TITLE
Placeholder doesn't appear with numeric IDs

### ISSUE THREAD

**Issue by ova2** [select2/select2#2980]
We often have numeric values in options. One examples how the native select is rendered in HTML:

```html
<select id="landWohnsitz" name="landWohnsitz" size="1"
    style="width:100%; display: none;" aria-required="true"
    data-placeholder="Bitte wählen ..." tabindex="-1">
    <option></option>
    <option value="80">Deutschland</option>
    <option value="87">Frankreich</option>
    <option value="83">Italien</option>
    <option value="85">Schweiz</option>
</select>
```

Placeholder doesn't appear in this case - neither with ```data-placeholder``` nor with ```placeholder``` JS option (don't matter how it's configured). String values are ok.

**Comment by kevin-brown** [comment]
Can you expand upon the issue you are facing by adding a few more pieces of information to your bug report?

- What steps can be used to reproduce the issue?
- What is the bug and what is the expected outcome?
- What browser(s) and Operating System have you tested with?
- Does the bug happen consistently across all tested browsers?
- What version of jQuery are you using? And what version of Select2?
- Are you using Select2 with other plugins?

The [contributing guide][cg] contains a useful section with a few questions which can help us track down the bug and speed up the process of finding a fix for it.  The creation of [an isolated test case][test-case] would also be useful.

[cg]: https://github.com/ivaynberg/select2/blob/master/CONTRIBUTING.md#reporting-bugs-with-select2
[test-case]: http://css-tricks.com/6263-reduced-test-cases/

**Comment by cawparky** [comment]
Hi, i have the same problem. If I give multiple select2 box into tags <form></form placeholder doesn't work (placeholder works outside of <form> tag fine ). I want get it to php(mysql) so it must work in <form>.
Please for some solution, here is my code.
  <form action="welcome.php" method="post">
		<div class="form-group" style="width:70px; height:44px;">
			<select id="multiple" class="form-control select2-multiple" multiple>
				<optgroup label="Alaskan/Hawaiian Time Zone">
	<option value="AK">Alaska</option>
	<option value="HI">Hawaii</option>
</optgroup>
<optgroup label="Pacific Time Zone">
	<option value="CA">California</option>
	<option value="NV">Nevada</option>
	<option value="OR">Oregon</option>
	<option value="WA">Washington</option>
                       </optgroup>
	      </select>
      </div>
      </form>
JS
----------

**Comment by ova2** [comment]
I had multiple select2 on a page, but placeholders works as long as values in options are not numbers.

My answeres to your questions:

What steps can be used to reproduce the issue?
Apply Select2 to any native select element with options like these
<option value="80">...</option>
<option value="81">...</option>

What is the bug and what is the expected outcome?
Configured placeholder (via HTML5 data-placeholder or JS option) doesn't appear in the Select2. I would expect a visible placeholder because it works with string values in options.

What browser(s) and Operating System have you tested with?
latest Chrome, Windows 7.

Does the bug happen consistently across all tested browsers?
Not tested all browsers, but yes.

What version of jQuery are you using? And what version of Select2?
jQuery 1.11.0, Select2 4,0 beta2

Are you using Select2 with other plugins?
Yes, sure, the page contains other plugins, but it doesn' matter

**Comment by ova2** [comment]
Do you need more information?

**Comment by kevin-brown** [comment]
@ova2 I still haven't got around to making a jsbin to reproduce this in.

I do hope to get this fixed (if I can reproduce this) by 4.0.0-rc.1.

**Comment by kevin-brown** [comment]
Unfortunately I was not able to reproduce this issue. I was also not able to reproduce the issue with Select2 within a `<form>`.

http://jsfiddle.net/46xnxuvo/2/

Tested on Chrome 39 on Ubuntu.

If someone can create a self-contained example that reproduces this issue you are seeing, I'll gladly reopen this ticket and figure out what the problem is.

Just a note: if you are using a custom `templateSelection` method, you have to remember that the placeholder is also put through the templating methods. That'll be noted in the templating documentation when it is created.
